### ISSUE TITLE
Поставщики: время работы

### ISSUE THREAD

**Issue by vitalk** [yandex-shri-minsk-2016/yummy-time#94]
Необходимо разобраться в каком формате задавать и хранить время работы поставщиков. Если сейчас для нас это не совсем важно, то позже, это позволит значительно упростить работу с сервисом.

С учётом того, что у нас интервал времени (режим работы), то это массив `[from, to]`. Где `from`/`to` строки. Всё это по дням недели

- либо массив массивов
  ```
  schedule: [
    [from, to],
    [from, to],
    [from, to],
    [from, to],
    [from, to],
    [from, to],
    [from, to]
  ]
  ```
  
- либо объект
  ```
  schedule: {
    1: [from, to],
    2: [from, to],
    3: [from, to],
    4: [from, to],
    5: [from, to],
    6: [from, to],
    7: [from, to]
  }
  ```

**Comment by criyology** [comment]
@vitalk, ты по сути написал одно и то же :)
Массив это и есть объект с ключами-цифрами. Предлагаю объект но с более понятными ключами типа `mon`, `tue` и так далее.

**Comment by octocrux** [comment]
А зачем нам вообще такие сложности с расписанием? К примеру, тот же pzz работает круглосуточно. Делать какие-то уведомления или запрещать пользователям заказывать еду в определенное время мы вроде не планируем.
По-моему, до тех пор, пока мы не решим как-то это расписание использовать, достаточно строки.

**Comment by vitalk** [comment]
Не все работают круглосуточно, у многих есть расписание доставки и при создании заказа это можно, _а потом и нужно_, будет учесть. Т.е. какой смысл давать возможность человеку создать заказ на 10-00, если служба доставки работает с 12-00.

**Comment by octocrux** [comment]
А какой смысл запрещать? Мы не оформляем заказ у поставщика автоматически, человек все равно должен позвонить и заказать еду.
Зато появится новый пласт проблем. А что, если поставщик решил поработать круглосуточно сегодня? А мы берем и запрещаем делать заказ. А что, если вообще решил не работать? Праздники тоже будем в расписание вносить? А если у поставщика изменится расписание и сдвинется на час?

**Comment by vitalk** [comment]
Согласен, запрещать плохо, но дать информацию о том что доставка работает тогда-то и тогда-то ещё на этапе создания заказа мне кажется не повредит.

**Comment by octocrux** [comment]
И с этим отлично справится простая строка текста.
Они же не отдел по гражданству и миграции :)

**Comment by criyology** [comment]
На самом деле, лучше облегчить пользователю жизнь и не заставлять сверяться с часами и лишний раз разочаровываться когда он голоден :)
Проблему с актуализацией расписания можно двумя способами: нанять девочку, которая за шоколадку будет ежедневно ходить по сайтам доставок и обновлять где надо информацию — или написать для этого робота, тем самым сэкономив кучу шоколадок.
